MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -875,6 +875,41 @@
 	_metalFeatures.minSwapchainImageCount = kMVKMinSwapchainImageCount;
 	_metalFeatures.maxSwapchainImageCount = kMVKMaxSwapchainImageCount;
 
+#if MVK_TVOS
+	_metalFeatures.mslVersionEnum = MTLLanguageVersion1_0;
+    _metalFeatures.maxPerStageTextureCount = 31;
+    _metalFeatures.mtlBufferAlignment = 64;
+	_metalFeatures.mtlCopyBufferAlignment = 1;
+    _metalFeatures.texelBuffers = true;
+	_metalFeatures.maxTextureDimension = (8 * KIBI);
+
+    if (supportsMTLFeatureSet(tvOS_GPUFamily1_v2)) {
+		_metalFeatures.mslVersionEnum = MTLLanguageVersion1_2;
```

Comment:
You should also check for `tvOS_GPUFamily1_v3` and set this to `MTLLanguageVersion2_0`, and `tvOS_GPUFamily1_v4` and set this to `MTLLanguageVersion2_1`.

Reply:
Appreciate the pointers here.